### PR TITLE
=htt #20014 Provide a description as stream size exception message

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #20014

### CODE REVIEW

**Comment by johanandren** [comment]
Recursive madness #20014 is this PR

The ticket fixed is #20013

**Comment by ktoso** [comment]
Fixes #20013 :wink:

File: akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerSpec.scala [hunk 1]
```diff
@@ -899,6 +899,7 @@ class HttpServerSpec extends AkkaSpec(
                   .thrownBy(entity.dataBytes.runFold(ByteString.empty)(_ ++ _).awaitResult(100.millis))
                   .getCause
                 error shouldEqual EntityStreamSizeException(limit, Some(actualSize))
+                error.getMessage should not be null
```
Comment (ktoso) [inline akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerSpec.scala]: `should include ("exceeded content length limit")`

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala [hunk 2]
```diff
@@ -80,7 +80,10 @@ object EntityStreamException {
  * The limit can also be configured in code, by calling [[HttpEntity#withSizeLimit]]
  * on the entity before materializing its `dataBytes` stream.
  */
-case class EntityStreamSizeException(limit: Long, actualSize: Option[Long] = None) extends RuntimeException {
+final case class EntityStreamSizeException(limit: Long, actualSize: Option[Long] = None) extends RuntimeException {
+
+  override def getMessage = toString
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala]: duh! Good catch, thanks.

**Comment by ktoso** [comment]
LGTM

**Comment by johanandren** [comment]
PLS BUILD

**Comment by viktorklang** [comment]
LGTM

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala [hunk 3]
```diff
@@ -80,7 +80,10 @@ object EntityStreamException {
  * The limit can also be configured in code, by calling [[HttpEntity#withSizeLimit]]
  * on the entity before materializing its `dataBytes` stream.
  */
-case class EntityStreamSizeException(limit: Long, actualSize: Option[Long] = None) extends RuntimeException {
+final case class EntityStreamSizeException(limit: Long, actualSize: Option[Long] = None) extends RuntimeException {
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala]: needs added mima filter:

```
[error]  * class akka.http.scaladsl.model.EntityStreamSizeException is declared final in current version
[error]    filter with: ProblemFilters.exclude[FinalClassProblem]("akka.http.scaladsl.model.EntityStreamSizeException")
```

**Comment by ktoso** [comment]
Needs mima fixed: https://github.com/akka/akka/pull/20014#discussion_r55973436 then merge please :+1:

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala (hunk 3 above)
Comment (johanandren) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala]: I wasn't sure, is adding final bc safe?

Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala]: Yes and no. Technically it is not, in practice no one should have been constructing this exception in the first place other than us. 

cc @patriknw

Comment (viktorklang) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala]: Technically, constructing is fine, extending is not :)

On Mon, Mar 14, 2016 at 10:40 AM, Konrad Malawski <notifications@github.com>
wrote:

> In akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala
> <https://github.com/akka/akka/pull/20014#discussion_r55974757>:
>
> > @@ -80,7 +80,10 @@ object EntityStreamException {
> >   * The limit can also be configured in code, by calling [[HttpEntity#withSizeLimit]]
> >   * on the entity before materializing its `dataBytes` stream.
> >   */
> > -case class EntityStreamSizeException(limit: Long, actualSize: Option[Long] = None) extends RuntimeException {
> > +final case class EntityStreamSizeException(limit: Long, actualSize: Option[Long] = None) extends RuntimeException {
>
> Yes and no. Technically it is not, in practice no one should have been
> constructing this exception in the first place other than us.
>
> cc @patriknw <https://github.com/patriknw>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/20014/files#r55974757>.
>



-- 
Cheers,
√

Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala]: Yeah, pre-coffee wording problems ;)

Comment (patriknw) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala]: technically it's even source incompatible, but I guess we not too strict with this new module

**Comment by patriknw** [comment]
LGTM

**Pull request merged**